src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultEnumConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class DefaultEnumConverter<TValue> : JsonValueConverter<TValue>
+    {
+        public bool TreatAsString { get; private set; }
+
+        internal DefaultEnumConverter(bool treatAsString)
+        {
+            TreatAsString = treatAsString;
+        }
+
+        public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TValue value)
+        {
+            if (TreatAsString)
+            {
+                // Assume the token is a string
+                if (reader.TokenType != JsonTokenType.String)
+                {
+                    value = default;
+                    return false;
+                }
+
+#if !BUILDING_INBOX_LIBRARY 
+                // todo: add code here to handle NS2.0
+                throw new NotImplementedException(SR.EnumConverterNotImplemented);
+#else
+                string enumString = reader.GetString();
+                if (!Enum.TryParse(valueType, enumString, out object objValue))
```

Comment:
The <TValue> isn't actually used for this enum converter since its <object>. However that's why the `Type valueType` is passed in - to get the underlying type. 

Does this help? Also there is an extensibility mechanism through an attribute (internal for now) that lets you write your own enum converter - both globally (all enums) and also for a specific enum type. This allows someone to do bit-wise \ flag enums and other formatting.

Reply:
You could make a `NonGenericEnumConverter` for types declared as `Enum` and `DefaultEnumConverter` could then start using it's generic parameter to avoid boxing the value. You then may want to cache the `DefaultEnumConverter`s per type in a static `ConcurrentDictionary` so each enum property doesn't instantiate it's own converter. Also, with things the way they currently are you'll probably need to add a `DefaultEnumNullableConverter` as well.